### PR TITLE
Bump version to 0.4.0-rc.1

### PR DESCRIPTION
**Pull request by r1viollet**
# What does this PR do?

Bump version to 0.4.0-rc.1

# Motivation

Creation of a new github release

# Additional Notes

The removal / update of the Cargo.lock file should be discussed

# How to test the change?

There are not significant changes here.

### CODE REVIEW

File: ddprof-exporter/Cargo.toml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 [package]
 name = "ddprof-exporter"
-version = "0.3.0"
+version = "0.4.0-rc.1"
```
Comment (sanchda) [inline ddprof-exporter/Cargo.toml:6]: Complete nit, but is tokenizing release candidates by `.` a convention?  If not, could I propose using a different delimiter than what we use for `major.minor.path` version numbers?  The reason is, it's nice for folks validating a version (potentially in various languages) when a delimiter separates tokens of uniform syntactical meaning.

Again, total nit.

**Review by sanchda — APPROVED** [review]
Looks good to me!

For future releases, it would be great to have a changelog we can agree on before it is generated in the release.  I'll make a note of this elsewhere.

File: ddprof-exporter/Cargo.toml (hunk 1 above)
Comment (morrisonlevi) [reply ddprof-exporter/Cargo.toml:6]: It's part of [semver](https://semver.org/), which is why I chose that originally. I imagine Erwan is just following what I did.

**Comment by morrisonlevi** [comment]
> For future releases, it would be great to have a changelog we can agree on before it is generated in the release. I'll make a note of this elsewhere.

Note that bumping this version doesn't automatically do a release. That's a separate thing. It can use autogenerated release notes from GitHub, which I've generally been doing (though sometimes curating).

**Review by morrisonlevi — APPROVED** [review]

File: ddprof-exporter/Cargo.toml (hunk 1 above)
Comment (r1viollet) [reply ddprof-exporter/Cargo.toml:6]: Yes. I will discuss releasing process at next status. If you have things in mind we would like to improve, feel free to mention these.

**Pull request merged**